Build failure: nodePackages_latest.nodejs

## Steps To Reproduce

Steps to reproduce the behavior:

1. build `nodePackages_latest.nodejs` from master
## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
┃        >   ...
┃        > ok 4059 known_issues/test-url-parse-conformance
┃        >   ---
┃        >   duration_ms: 56.50900
┃        >   ...
┃        > ok 4060 known_issues/test-vm-function-declaration-uses-define
┃        >   ---
┃        >   duration_ms: 36.65000
┃        >   ...
┃        > ok 4061 known_issues/test-vm-timeout-escape-nexttick
┃        >   ---
┃        >   duration_ms: 806.78800
┃        >   ...
┃        >
┃        > Failed tests:
┃        > out/Release/node /build/node-v23.2.0/test/parallel/test-os.js
┃        > make: *** [Makefile:558: test-ci-js] Error 1
┃        For full logs, run 'nix log /nix/store/ahgdyngidcphsmb7zn5b1lny589pyjdc-nodejs-23.2.0.drv'.
```

</details>

## Additional context

Found it while building `zed-editor` on master.

## Metadata

❯  nix-info -m                                                                                 
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.7-cachyos, NixOS, 24.11 (Vicuna), 24.11.20241109.76612b1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/gg86rfp39vc7chqsszk32q7995hz4943-source`


## Notify maintainers

@aduh95 @Ma27 

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/web/nodejs/v23.nix
+++ b/pkgs/development/web/nodejs/v23.nix
@@ -53,5 +53,15 @@ buildNodejs {
       hash = "sha256-gmIyiSyNzC3pClL1SM2YicckWM+/2tsbV1xv2S3d5G0=";
       revert = true;
     })
+    # Fix for https://github.com/NixOS/nixpkgs/issues/355919
+    # FIXME: remove after a minor point release
+    (fetchpatch2 {
+      url = "https://github.com/nodejs/node/commit/a094a8166cd772f89e92b5deef168e5e599fa815.patch?full_index=1";
+      hash = "sha256-5FZfozYWRa1ZI/f+e+xpdn974Jg2DbiHbua13XUQP5E=";
+    })
+    (fetchpatch2 {
+      url = "https://github.com/nodejs/node/commit/f270462c09ddfd770291a7c8a2cd204b2c63d730.patch?full_index=1";
+      hash = "sha256-Err0i5g7WtXcnhykKgrS3ocX7/3oV9UrT0SNeRtMZNU=";
+    })
   ];
 }

